src/commands/createNewProject/createNewProject.ts

```diff
@@ -55,20 +55,31 @@ export async function createNewProject(
     }
     actionContext.properties.projectLanguage = language;
 
-    const projectCreator: ProjectCreatorBase = getProjectCreator(language, functionAppPath, actionContext);
-    await projectCreator.addNonVSCodeFiles();
+    await window.withProgress({ location: ProgressLocation.Notification, title: localize('creating', 'Creating new project...') }, async () => {
+        // tslint:disable-next-line:no-non-null-assertion
+        functionAppPath = functionAppPath!;
```

Comment:
Interesting solution...

Reply:
Added your nonNull stuff in this PR: https://github.com/Microsoft/vscode-azurefunctions/pull/607

Will probably use that going forward